File: old/basic/src/main/java/com/dinesh/android/java/fragment/ThirdFragment.java
package com.dinesh.android.java.fragment;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentResultListener;
import androidx.fragment.app.FragmentTransaction;

import com.dinesh.android.R;

public class ThirdFragment extends Fragment {
    private final String TAG = "log_" + getClass().getName().split(getClass().getName().split("\\.")[2] + ".")[1];

    TextView textView;
    EditText et_ThirdFragment;
    Button btnSecondFragment, btnFirstFragment;

    String result;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_third,container,false);

        textView = view.findViewById(R.id.textView);
        et_ThirdFragment = view.findViewById(R.id.et_ThirdFragment);
        btnSecondFragment = view.findViewById(R.id.btnSecondFragment);
        btnFirstFragment = view.findViewById(R.id.btnFirstFragment);

        FragmentTransaction fragmentTransaction = requireActivity().getSupportFragmentManager().beginTransaction();

        btnSecondFragment.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                result = et_ThirdFragment.getText().toString();
                Log.e(TAG, "onClick: SecondFragment");
                Toast.makeText(getActivity(), "Hello SecondFragment", Toast.LENGTH_SHORT).show();
                fragmentTransaction.replace(R.id.fragment, new SecondFragment());
                fragmentTransaction.commit();
            }
        });

        btnFirstFragment.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                result = et_ThirdFragment.getText().toString();
                Log.e(TAG, "onClick: ThirdFragment");
                Toast.makeText(getActivity(), "Hello FirstFragment", Toast.LENGTH_SHORT).show();
                fragmentTransaction.replace(R.id.fragment, new FirstFragment());
                fragmentTransaction.commit();
            }
        });

        getParentFragmentManager().setFragmentResultListener("requestKey", this, new FragmentResultListener() {
            @Override
            public void onFragmentResult(@NonNull String requestKey, @NonNull Bundle bundle) {
                // We use a String here, but any type that can be put in a Bundle is supported
                result = bundle.getString("bundleKey");
                et_ThirdFragment.setText(result);
                Log.i(TAG, "onFragmentResult: " + result);
            }
        });

        return view;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        Log.e(TAG, "onDestroyView: " );
        Bundle bundle = new Bundle();
        bundle.putString("bundleKey", result);
        getParentFragmentManager().setFragmentResult("requestKey", bundle);
    }

}

File: old/basic/src/main/java/com/dinesh/android/kotlin/fragment/ThirdFragment.kt
package com.dinesh.android.kotlin.fragment

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import androidx.fragment.app.setFragmentResult
import androidx.fragment.app.setFragmentResultListener
import androidx.navigation.fragment.findNavController
import com.dinesh.android.R
import com.dinesh.android.databinding.FragmentThirdBinding


class ThirdFragment : Fragment() {
    private val TAG = "log_" + ThirdFragment::class.java.name.split(ThirdFragment::class.java.name.split(".").toTypedArray()[2] + ".").toTypedArray()[1]
    private var _binding: FragmentThirdBinding? = null
    private val binding get() = _binding!!
    lateinit var result: String

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        _binding = FragmentThirdBinding.inflate(inflater, container, false)
        Log.d(TAG, "onCreateView: ")

        passDataBetweenFragment()

        return binding.root
    }

    private fun passDataBetweenFragment() {
        binding.btnSecondFragment.setOnClickListener {
            result = binding.etThirdFragment.text.toString()
            Log.d(TAG, "btnSecondFragment: ${result}")
            findNavController().navigate(R.id.action_thirdFragment_to_secondFragment)
        }

        binding.btnFirstFragment.setOnClickListener {
            result = binding.etThirdFragment.text.toString()
            Log.d(TAG, "btnFirstFragment: ${result}")
            findNavController().navigate(R.id.action_thirdFragment_to_firstFragment)
        }

        setFragmentResultListener("requestKey") { requestKey, bundle ->
            result = bundle.getString("bundleKey").toString()
            Log.e(TAG, "setFragmentResultListener: ${result}")
            binding.etThirdFragment.setText(result)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        Log.e(TAG, "onDestroyView: ")

        //  passDataBetweenFragment
        setFragmentResult("requestKey", bundleOf("bundleKey" to result))

        _binding = null
    }


}